influxdb_client_3/__init__.py

```diff
@@ -194,10 +216,15 @@ def query(self, query, language="sql", mode="all", database=None,**kwargs ):
             database = self._database
         
         try:
-            headers = [(b"authorization", f"Bearer {self._token}".encode('utf-8'))]
-    
             # Create an authorization header
-            _options = FlightCallOptions(headers=headers, **kwargs)
+            optargs = {
+                "headers": [(b"authorization", f"Bearer {self._token}".encode('utf-8'))],
+                "timeout": 300
+            }
+            opts = self._merge_options(optargs, kwargs)
```

Comment:
Hi @jstirnaman, just reviewing again. I am just trying to simplify this a little for the purpose of performance.  In most cases, users will not modify the header unless told to do so. So could we perhaps change it so there is conditional logic on whether we call the merge function. If `headers=None`  we should go with the default. In its current form, we are statically defining that the only two options allowed are
```
            _options = FlightCallOptions(
                headers=opts["headers"],
                timeout=opts["timeout"])
```
Which means any other FlightCallOption is dropped. https://arrow.apache.org/docs/python/generated/pyarrow.flight.FlightCallOptions.html
Could we have it so there is a headers  option when the user calls a query which takes:
`headers=[(b"influx-debug-id", b"JasTraceId")]`
A bit like you have in your example. That way we only perform the merge on the headers if they exist and allow the flight options to be applied as normal. Thanks again for your work :pray: